src/lexer.coffee

```diff
@@ -369,18 +369,31 @@ exports.Lexer = class Lexer
           comment
         .filter (comment) -> comment
 
+    getIndentSize = ({leadingWhitespace, nonInitial}) ->
+      lastNewlineIndex = leadingWhitespace.lastIndexOf '\n'
+      if hereComment? or not nonInitial
+        return null unless lastNewlineIndex > -1
+      else
+        lastNewlineIndex ?= -1
+      leadingWhitespace.length - 1 - lastNewlineIndex
     offsetInChunk = 0
     commentAttachments = for {content, length, leadingWhitespace, precededByBlankLine}, i in contents
       nonInitial = i isnt 0
       leadingNewlineOffset = if nonInitial then 1 else 0
       offsetInChunk += leadingNewlineOffset + leadingWhitespace.length
+      indentSize = getIndentSize {leadingWhitespace, nonInitial}
+      noIndent = not indentSize? or indentSize is -1
       commentAttachment = {
         content
         here: hereComment?
         newLine: leadingNewline or nonInitial # Line comments after the first one start new lines, by definition.
         locationData: @makeLocationData {offsetInChunk, length}
         precededByBlankLine
+        indentSize
```

Comment:
We are giving the individual comments "indent-awareness" that can then be used by the rewriter (when determining how to adjust `INDENT`/`OUTDENT` location data)